packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,226 @@
+    /// @dev MUST set challenge_start_time to the current block time
+    /// @dev MUST emit the ChallengeStarted event
+    /// @param _safe The Safe to challenge
+    function challenge(address _safe) external {
+        ModuleConfig storage config = safeConfigs[_safe];
+
+        if (config.fallbackOwner == address(0)) {
+            revert LivenessModule2_ModuleNotConfigured();
+        }
+
+        // Check that the module is still enabled on the target Safe
+        Safe safe = Safe(payable(_safe));
+        if (!safe.isModuleEnabled(address(this))) {
+            revert LivenessModule2_ModuleNotEnabled();
+        }
+
+        if (msg.sender != config.fallbackOwner) {
+            revert LivenessModule2_UnauthorizedCaller();
+        }
+
+        if (challengeStartTime[_safe] != 0) {
+            revert LivenessModule2_ChallengeAlreadyExists();
+        }
+
+        challengeStartTime[_safe] = block.timestamp;
+        emit ChallengeStarted(_safe, block.timestamp);
+    }
+
+    /// @notice Responds to a challenge for an enabled safe, canceling it
+    /// @dev MUST only be executable by an enabled safe
+    /// @dev MUST revert if there isn't a challenge for the calling safe
+    /// @dev MUST revert if there is a challenge for the calling safe but the response period has expired
+    /// @dev MUST emit the ChallengeCancelled event
+    function respond() external {
+        // Check that this module is enabled on the calling Safe
+        Safe safe = Safe(payable(msg.sender));
+        if (!safe.isModuleEnabled(address(this))) {
+            revert LivenessModule2_ModuleNotEnabled();
+        }
+
+        ModuleConfig storage config = safeConfigs[msg.sender];
+
+        // Check if the calling safe has configuration set
+        if (config.fallbackOwner == address(0)) {
+            revert LivenessModule2_ModuleNotConfigured();
+        }
+
+        uint256 startTime = challengeStartTime[msg.sender];
+        if (startTime == 0) {
+            revert LivenessModule2_ChallengeDoesNotExist();
+        }
+
+        // Check if response period has expired
+        if (block.timestamp >= startTime + config.livenessResponsePeriod) {
+            revert LivenessModule2_ResponsePeriodEnded();
+        }
+
+        delete challengeStartTime[msg.sender];
+        emit ChallengeCancelled(msg.sender);
+    }
+
+    /// @notice With a successful challenge, removes all current owners from an enabled safe, appoints fallback as its
+    /// sole owner, and sets its quorum to 1
+    /// @dev MUST be executable by anyone
+    /// @dev MUST revert if the given safe hasn't enabled the module
+    /// @dev MUST revert if there isn't a successful challenge for the given safe
+    /// @dev MUST enable the module to start a new challenge
+    /// @dev MUST emit the ChallengeExecuted event
+    /// @param _safe The Safe to transfer ownership of
+    function changeOwnershipToFallback(address _safe) external {
```

Comment:
We should require msg.sender == fallback owner. There's no real reason *not* to make this assertion, it fits well into the expected behavior, and it provides better guarantees around when this function will be called.

Reply:
Fixed in e03a5d7dd728851d60cf929827f0cbec5f1803ee